GitForWindowsHelper/index.js

```diff
@@ -83,6 +83,17 @@ module.exports = async function (context, req) {
         return withStatus(500, undefined, e.message || JSON.stringify(e, null, 2))
     }
 
+    try {
+        const {closeReleaseMilestone} = require('./update-milestones')
+        if (req.headers['x-github-event'] === 'pull_request'
+            && req.body.repository.full_name === 'git-for-windows/git'
+            && req.body.action === 'closed'
+            && req.body.pull_request.merged === 'true') return ok(await closeReleaseMilestone(context, req))
```

Comment:
Alternatively, we could piggy-back onto [the code that "pushes" the PR branch to `main`](https://github.com/git-for-windows/gfw-helper-github-app/blob/73e1cd9e9b5faea291e240fc8ea4b0d3581bb10a/GitForWindowsHelper/finalize-g4w-release.js#L56-L59). That would cut out one extra round trip between GitHub and the GitForWindowsHelper, and simplify the mental model of the process.

Reply:
It does cut out a roundtrip, but it does require duplication of the "is this a pre-release" logic from https://github.com/git-for-windows/git-for-windows-automation/blob/main/.github/actions/github-release/action.yml and maybe some special handling of out of band releases, whereas the `release.released` approach would get that for free.